### PR TITLE
Adding posts flow annotations

### PR DESCRIPTION
**Pull request by chrux**
#### Summary
Migrate to Flow file src/actions/posts.js in mattermost-redux

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-11046

Fixes https://github.com/mattermost/mattermost-server/issues/9771

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed

#### Test Information
This PR was tested on: Macbook Pro, MacOS High Sierra 10.13.6

This completes this issue https://github.com/mattermost/mattermost-server/issues/9771

### CODE REVIEW

File: src/actions/posts.js [hunk 1]
```diff
@@ -951,6 +967,8 @@ export async function getProfilesAndStatusesForPosts(posts, dispatch, getState)
     const statusesToLoad = new Set();
 
     Object.values(posts).forEach((post) => {
+        // https://github.com/facebook/flow/issues/2221
+        // $FlowFixMe - Object.values currently does not have good flow support
```
Comment (hmhealey) [inline src/actions/posts.js]: I ran into this exact same issue when trying to annotate this function on one of our feature branches, even when I had the object as `{[string]: Post}`. It's a tough one

File: package-lock.json [hunk 2]
```diff
@@ -22,1585 +22,940 @@
...
-        "resolve": "^1.3.2",
-        "semver": "^5.4.1",
-        "source-map": "^0.5.0"
-      },
-      "dependencies": {
-        "@babel/code-frame": {
-          "version": "7.0.0",
-          "resolved": "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.0.0.tgz",
-          "integrity": "sha512-OfC2uemaknXr87bdLUkWog7nYuliM9Ij5HUcajsVcMCpQrcLmtxRbVFTIqmcSkSeYRBFBRxs2FiUqFJDLdiebA==",
+        "anymatch": {
```
Comment (hmhealey) [inline package-lock.json]: Can you revert the changes to this file? I don't think they should be changing from this PR. You might also want to update your versions of npm/node since the newest version shouldn't cause this many unintentional changes

**Comment by hmhealey** [comment]
There's also a unit test failure, but I can't see a reason for it since it's a different part of the code. I'm going to trigger a rebuild to see if it fixes itself, but I'll keep an eye on it to see if it keeps failing.

**Comment by sudheerDev** [comment]
Thanks @jespino for jumping in the review for me :)

File: src/actions/posts.js (hunk 1 above)
Comment (jespino) [reply src/actions/posts.js]: Yes, but I think there are workarounds to this problem. I think you can do something like:

```javascript
Object.keys(posts).forEach((key: string) => {
    const userId = posts[key].user_id;
    ...
});
```

File: src/actions/posts.js [hunk 3]
```diff
@@ -1116,17 +1142,17 @@ export function selectPost(postId) {
     };
 }
 
-export function selectFocusedPostId(postId) {
+export function selectFocusedPostId(postId: string) {
```
Comment (jespino) [inline src/actions/posts.js]: I miss here the return type which is an `Action`

File: src/types/actions.js [hunk 4]
```diff
@@ -20,7 +20,12 @@ type BatchAction = {
     payload: Array<GenericAction>;
 }
 
-type Action = GenericAction | Thunk | BatchAction
+export type PostAction = {
+    ...GenericAction,
+    postId?: string
```
Comment (jespino) [inline src/types/actions.js]: This looks like an exception only used in the `RECEIVED_FILES_FOR_POST` action. Maybe I would prefer a explicit `ReceivedFilesForPostAction` with the definition similar to:

```javascript
{|
    type:FileTypes.RECEIVED_FILES_FOR_POST,
    postId: string,
    data: Array<FileInfo>,
|}
```

**Comment by chrux** [comment]
Will do it this weekend

**Comment by grundleborg** [comment]
Hey @chrux , just wondering if you've had a chance to take a look at the changes needed on this PR yet? Let us know if there's anything we can do to help.

**Comment by hanzei** [comment]
Hey @chrux,

Do you have any questions about the provided feedback? If you have questions feel free to PM me on the [Mattermost Community Server](https://community.mattermost.com).

Please note that we will close this PR after another 30 days of inactivity. If you don't object, contributors might complete this PR for you.

**Comment by hanzei** [comment]
This PR has become orphaned and will be closed. Still, thanks for your contribution @chrux
